run/test/specs/message_checks_groups_android.spec.ts

```diff
@@ -0,0 +1,546 @@
+  const { device1, device2, device3 } = await openAppThreeDevices(platform);
+  // Create users A, B and C
+  const [userA, userB, userC] = await Promise.all([
+    newUser(device1, "Alice", platform),
+    newUser(device2, "Bob", platform),
+    newUser(device3, "Charlie", platform),
+  ]);
+  // Create contact between User A and User B
+  await createGroup(
+    platform,
+    device1,
+    userA,
+    device2,
+    userB,
+    device3,
+    userC,
+    testGroupName
+  );
+  const replyMessage = `Replying to document from ${userA.userName} in ${testGroupName}`;
+  await device1.clickOnElement("Attachments button");
+  await sleepFor(100);
+  await device1.clickOnElement("Documents folder");
+  const mediaButtons = await device1.findElementsByClass(
+    "android.widget.CompoundButton"
+  );
+  const documentsButton = await device1.findMatchingTextInElementArray(
+    mediaButtons,
+    "Documents"
+  );
+  if (!documentsButton) {
+    throw new Error("documentsButton was not found");
+  }
+  await device1.click(documentsButton.ELEMENT);
+  const testDocument = await device1.doesElementExist({
+    strategy: "id",
+    selector: "android:id/title",
+    maxWait: 1000,
+    text: "test_file.pdf",
+  });
+  if (!testDocument) {
+    await runScriptAndLog(
+      `adb -s emulator-5554 push 'run/test/specs/media/test_file.pdf' /storage/emulated/0/Download`,
+      true
+    );
+  }
+  await sleepFor(1000);
+  await device1.clickOnTextElementById("android:id/title", "test_file.pdf");
+  // Reply to message
+  await sleepFor(1000);
+  await device2.waitForTextElementToBePresent({
+    strategy: "accessibility id",
+    selector: "Document",
+  });
+  await device3.waitForTextElementToBePresent({
+    strategy: "accessibility id",
+    selector: "Document",
+  });
+  await device2.longPress("Document");
+  await device2.clickOnElement("Reply to message");
+  await device2.sendMessage(replyMessage);
+  await device1.waitForTextElementToBePresent({
+    strategy: "accessibility id",
+    selector: "Message body",
+    text: replyMessage,
+  });
+  await device3.waitForTextElementToBePresent({
+    strategy: "accessibility id",
+    selector: "Message body",
+    text: replyMessage,
+  });
+  // Close app and server
+  await closeApp(device1, device2, device3);
+}
+
+async function sendLinkGroup(platform: SupportedPlatformsType) {
+  const testGroupName = "Message checks for groups";
+  const { device1, device2, device3 } = await openAppThreeDevices(platform);
+  // Create users A, B and C
+  const [userA, userB, userC] = await Promise.all([
+    newUser(device1, "Alice", platform),
+    newUser(device2, "Bob", platform),
+    newUser(device3, "Charlie", platform),
+  ]);
+  // Create contact between User A and User B
+  await createGroup(
+    platform,
+    device1,
+    userA,
+    device2,
+    userB,
+    device3,
+    userC,
+    testGroupName
+  );
+  const testLink = `https://nerdlegame.com/`;
```

Comment:
same

Reply:
the reason we don't want to use nerdle.com is because they might change the title without telling us and our test will start to break for no reasons. And they might ban use as we would be making too many requests in CI

Reply:
Ok have updated to `https://example.net/`